pkg/apis/networking/validation/validation.go

```diff
@@ -32,6 +31,12 @@ import (
 	"k8s.io/kubernetes/pkg/apis/networking"
 )
```

Comment:
Is there any sort of validation that we can do around the IngressClass type? For example does it make sense to validate that you DON'T/DO have both an IngressClass and Annotation at the same time?

Reply:
I don't think we should have a mutually exclusion rule of annotation and IngressClass property as it will likely be violated during the migration.

We have discussed this before that we will define the precedence order between the two.

Reply:
There is conversion that should automatically translate the value of the ingress class annotation to the new field. I think that means that consumers of the v1 API should be able to rely on the class field exclusively (annotations from previous API versions would be converted). I'm not sure that we need to add validation to prevent an annotation being sent, but I do think that at least documentation should be clear about the field that should be used.